Core/Operator/Slots/Slot.cs

```diff
@@ -93,16 +94,15 @@ public T GetValue(EvaluationContext context)
 
         public void AddConnection(ISlot sourceSlot, int index = 0)
         {
-            if (!IsConnected && sourceSlot != null)
+            Debug.Assert(sourceSlot is not null);
```

Comment:
Interesting! Can you make this assumption? I didn't check the code...

Reply:
yes, this is the only place it is called:

```
public bool AddConnection(Symbol.Connection connection, int multiInputIndex)
        {
            var foundInstance = TryGetInstancesForConnection(connection, out _, out var sourceSlot, out _, out var targetSlot);

            if (!foundInstance)
                return false;
            
            targetSlot.AddConnection(sourceSlot, multiInputIndex);
            return true;
        }
```